### DISCUSSION TITLE
How to match values in CDB list ?

### DISCUSSION THREAD

**Discussion by swapug** [wazuh/wazuh#14347] [Q&A] [answered]
I have created a CDB list of blacklisted countries, but as some of the countries have spaces in them they cannot be used as key. So how do I use value to match or add spaces in a key? The mentioned documentation says to use check_value to match against the value but I cannot match every single value in the rule.

**Comment by Selutario** [comment, marked as answer]
Hi @swapug,

I just tried it and it is actually possible to use words with spaces as a key in the CDB list. In fact, it was not even necessary in my case to put it in quotes. This is a little example:

```
# cat /var/ossec/etc/lists/test-countries
test:value
test2:value2
test3:value3
United Kingdom:value4
test5:value5
```

Decoder:
```XML
<decoder name="example">
  <program_name>^example</program_name>
</decoder>

<decoder name="example">
  <parent>example</parent>
  <regex>User '(\w+)' logged from '(\.*)'</regex>
  <order>user, country</order>
</decoder>
```

Rule:
```XML
<rule id="100010" level="0">
  <program_name>example</program_name>
  <description>User logged</description>
  <list field="country" lookup="match_key">etc/lists/test-countries</list>
</rule>
```

This is the wazuh-logtest output:
```
# /var/ossec/bin/wazuh-logtest
Starting wazuh-logtest v4.3.0
Type one log per line

Dec 25 20:45:02 MyHost example[12345]: User 'admin' logged from 'United Kingdom'

**Phase 1: Completed pre-decoding.
	full event: 'Dec 25 20:45:02 MyHost example[12345]: User 'admin' logged from 'United Kingdom''
	timestamp: 'Dec 25 20:45:02'
	hostname: 'MyHost'
	program_name: 'example'

**Phase 2: Completed decoding.
	name: 'example'
	country: 'United Kingdom'
	dstuser: 'admin'

**Phase 3: Completed filtering (rules).
	id: '100010'
	level: '0'
	description: 'User logged'
	groups: '['local', 'syslog', 'sshd']'
	firedtimes: '1'
	mail: 'False'



Dec 25 20:45:02 MyHost example[12345]: User 'admin' logged from 'Norway'

**Phase 1: Completed pre-decoding.
	full event: 'Dec 25 20:45:02 MyHost example[12345]: User 'admin' logged from 'Norway''
	timestamp: 'Dec 25 20:45:02'
	hostname: 'MyHost'
	program_name: 'example'

**Phase 2: Completed decoding.
	name: 'example'
	country: 'Norway'
	dstuser: 'admin'
```

It is possible that the problem is in your decoder, which could be capturing only "word" characters and not the full name in those cases that contain spaces. If possible for your use case, use `\.*` to capture the country name or some alternative regex. You can take a look at the syntax here:
- [Regular Expression Syntax - Wazuh](https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/regex.html)
